### PR TITLE
Initial version of the Vert.x Generator

### PR DESCRIPTION
**Pull request by cescoffier**
A very simple generator for Vert.x project.

### CODE REVIEW

**Comment by iocanel** [comment]
great minds think alike: #664 :-)

File: generator/vertx/src/main/java/io/fabric8/maven/generator/vertx/VertxGenerator.java [hunk 1]
```diff
@@ -16,32 +16,65 @@
...
+  public boolean isApplicable(List<ImageConfiguration> configs) throws MojoExecutionException {
+    return shouldAddImageConfiguration(configs)
+        && (MavenUtil.hasPlugin(getProject(), VERTX_MAVEN_PLUGIN_GA)
+        || MavenUtil.hasDependency(getProject(), VERTX_GROUPID));
+  }
+
+  @Override
+  protected Map<String, String> getEnv(boolean isPrePackagePhase) throws MojoExecutionException {
+    Map<String, String> ret = super.getEnv(isPrePackagePhase);
+    ret.put("JAVA_OPTIONS", "-Dvertx.cacheDirBase=/tmp");
```
Comment (rhuss) [inline generator/vertx/src/main/java/io/fabric8/maven/generator/vertx/VertxGenerator.java]: might be a good idea to examine whether the parent already added a `JAVA_OPTIONS` and possibly add this to an already existing entry.

**Comment by cescoffier** [comment]
I've fixed the JAVA_OPTIONS thingy. It should also be fixed in the Spring Boot generator (copied it from there ;-))

**Comment by rhuss** [comment]
Yeah, will fix that there, too. 

Thanks !

[merge]

**Pull request merged**